istio-operator/istio-operator.spec

```diff
@@ -84,6 +84,10 @@ popd
 cp -r src/github.com/maistra/istio/install/kubernetes/helm/ src/github.com/maistra/istio-operator/tmp/_output
 pushd src/github.com/maistra/istio-operator/
 COMMUNITY=%{community_build} MAISTRA_VERSION=%{version} SOURCE_DIR=. HELM_DIR=./tmp/_output/helm ./tmp/build/patch-charts.sh
+popd
+
+# update image name and version
+sed -i -e 's/image: maistra\/istio-operator.*/image: maistra\/istio-operator-ubi8:1.0.0/' src/github.com/maistra/istio-operator/manifests/1.0.0/maistra.v1.0.0.clusterserviceversion.yaml
```

Comment:
Ideally these should be changed by the build process. I understand that having a Makefile would be easier to achieve that, so, we can improve this later. Could you please create a Jira for that and put a TODO item here so we can look at this in the future?

Reply:
To clarify, I'm thinking on having an env variable that the build process (make? build.sh?) would read and patch the appropriate file. Then, each .spec (community or product) would set an different value for it, with no need to apply a sed in the .spec file.

Reply:
This might be the place to have this conversation:

I could generate all of the files under "1.0.0" based on a csv template and one of the yaml files in istio-operator's "/deploy" dir. Things like image name, istio-operator role permissions can be copied verbatim (yq makes it pretty easy to do too). 

At this point I don't know how much difference there will be between the community and product versions of the metadata (description fields, the icon), worst comes to worst I could have two separate templates. 

Then during the build we could generate metadata, and package the appropriate version (community or product) in the rpm, with no subsequent changes required.

The main downside is that if this was to be done during rpm build, we'll need to add a dependency on jq and yq (I don't want to parse yaml via regexes). I don't see yq package in brew, but we could convert our yamls in deploy/ to json, and we'd depend on jq only, which is available in brew.

Also @brian-avery, @rcernich.

Reply:
I'm more in favor of doing that on build time. That's how most projects deal with that, generating docs, man pages and other stuff that are not binary executables (/me remembers the old good days of autotools :))

I don't understand what you say about having two templates in the worst case. Aren't templates mean to be customizable, having variables in them so that you can generate many artifacts from one single template?

Also, this dependency on yq is bad. Even jq is bad for us because it's only available in ubi8 with a proper subscription on the host, but we are using it anyway for cni already, so, it's not a blocker.

If I understand correctly, yq is just a wrapper on top of jq, converting from yaml to json before invoking jq, correct? Can we do the same at runtime, i.e., without having to change our files in deploy/?

Reply:
Removed that bit altogether, will generate metadata separately